src/main/java/com/hrznstudio/galacticraft/block/special/TinLadderBlock.java

```diff
@@ -0,0 +1,211 @@
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package com.hrznstudio.galacticraft.block.special;
+
+import net.fabricmc.api.EnvType;
+import net.fabricmc.api.Environment;
+import net.minecraft.block.*;
+import net.minecraft.entity.player.PlayerEntity;
+import net.minecraft.fluid.FluidState;
+import net.minecraft.fluid.Fluids;
+import net.minecraft.item.ItemPlacementContext;
+import net.minecraft.item.ItemStack;
+import net.minecraft.state.StateManager;
+import net.minecraft.state.property.BooleanProperty;
+import net.minecraft.state.property.DirectionProperty;
+import net.minecraft.state.property.Properties;
+import net.minecraft.util.ActionResult;
+import net.minecraft.util.BlockMirror;
+import net.minecraft.util.BlockRotation;
+import net.minecraft.util.Hand;
+import net.minecraft.util.hit.BlockHitResult;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.util.math.Direction;
+import net.minecraft.util.shape.VoxelShape;
+import net.minecraft.util.shape.VoxelShapes;
+import net.minecraft.world.BlockView;
+import net.minecraft.world.World;
+import net.minecraft.world.WorldAccess;
+import net.minecraft.world.WorldView;
+import org.jetbrains.annotations.Nullable;
+
+/**
+ * @author <a href="https://github.com/StellarHorizons">StellarHorizons</a>
+ */
+public class TinLadderBlock extends Block {
+
+    public static final DirectionProperty FACING;
+    public static final BooleanProperty WATERLOGGED;
+    protected static final VoxelShape EAST_SHAPE;
+    protected static final VoxelShape WEST_SHAPE;
+    protected static final VoxelShape SOUTH_SHAPE;
+    protected static final VoxelShape NORTH_SHAPE;
+
+    public TinLadderBlock(Settings settings) {
+        super(settings);
+        this.setDefaultState(this.stateManager.getDefaultState()
+                .with(FACING, Direction.NORTH)
+                .with(WATERLOGGED, false));
+    }
+
+    @Nullable
+    private ActionResult checkCanTinLadderBePlaced(World world, BlockPos checkPos, BlockState state) {
+        if (world.getBlockState(checkPos).isAir()) {
+            BlockState newState = this.getDefaultState().with(FACING, state.get(FACING));
+            world.setBlockState(checkPos, newState);
+            return ActionResult.SUCCESS;
+        } else if (!(world.getBlockState(checkPos).getBlock() instanceof TinLadderBlock)) {
+            return ActionResult.PASS;
+        }
+        return null;
```

Comment:
Maybe just return `ActionResult.PASS`? You check for null later and return `ActionResult.PASS` if it is later.

Reply:
Added, thanks!